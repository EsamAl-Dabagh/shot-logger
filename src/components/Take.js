import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';
import Switch from '@material-ui/core/Switch';

class Take extends Component {
  constructor() {
    super();

    this.state = {
      scene: "",
      shot: "",
      takeNumber: "",
      description: "",
      goodTake: false,
      notes: ""
    }
  }

  render() {
    return (
      <div className="take">
        <p>This is the TAKE component: {this.props.id}</p>
        <form>
          <TextField 
            className="scene-input" 
            variant="outlined" 
            label="Scene"
            autoFocus
            onChange={event => this.setState({ scene: event.target.value })} 
          />
          <TextField 
            className="shot-input"
            variant="outlined"
            label="Shot"
            onChange={event => this.setState({ shot: event.target.value })} 
          />
          <TextField 
            className="take-input"
            variant="outlined" 
            label="Take"
            onChange={event => this.setState({ takeNumber: event.target.value })} 
          />
          <TextField 
            className="desc-input"
            variant="outlined" 
            label="Description"
            onChange={event => this.setState({ description: event.target.value })} 
          />
              
          <Switch 
            className="goodTake-switch"
            checked={this.state.goodTake}
            onChange={event => this.setState({ goodTake: event.target.checked })} 
            value="goodTake"
            color="primary"
            label="Good Take?"
          />
          
          <TextField
            className="notes-input"
            variant="outlined"
            label="Notes"
            multiline
            onChange={event => this.setState({ notes: event.target.value })}
          />
        </form>
      </div>
      
    )
  }
}

export default Take;