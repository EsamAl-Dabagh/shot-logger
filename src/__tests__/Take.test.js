import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Take from '../components/Take';

describe("Take", () => {

  const mockRemove = jest.fn();
  const id = 1;
  const key = 1;
  const props = { key: key, id: id, removeTake: mockRemove }
  const take = shallow(<Take {...props} />);

  it("renders correctly", () => {
    expect(take).toMatchSnapshot();
  });

  it("initializes with scene, shot, takeNumber, description, goodTake, and notes in `state`", () => {
    expect(take.state()).toEqual({
      scene: "",
      shot: "",
      takeNumber: "",
      description: "",
      goodTake: false,
      notes: ""
    });
  });

  describe("when typing into the Scene input", () => {
    it("updates the scene in `state`", () => {
      take.find(".scene-input").simulate("change", { target: { value: "1" } });
      expect(take.state().scene).toEqual("1");
    });
  });

  describe("when typing into the Shot input", () => {
    it("updates the shot in `state`", () => {
      take.find(".shot-input").simulate("change", { target: { value: "1" } });
      expect(take.state().shot).toEqual("1");
    });
  });

  describe("when typing into the Take input", () => {
    it("updates the takeNumber in `state`", () => {
      take.find(".take-input").simulate("change", { target: { value: "1" } });
      expect(take.state().takeNumber).toEqual("1");
    });
  });

  describe("when typing into the Description input", () => {
    it("updates the description in `state`", () => {
      take.find(".desc-input").simulate("change", { target: { value: "A description" } });
      expect(take.state().description).toEqual("A description");
    });
  });

  describe("when Good Take switch is clicked", () => {
    it("updates goodTake in `state`", () => {
      take.find(".goodTake-switch").simulate("change", { target: { checked: true } });
      expect(take.state().goodTake).toBe(true);
    });
  });

  describe("when typing into the Notes input", () => {
    it("updates the notes in `state`", () => {
      take.find(".notes-input").simulate("change", { target: { value: "Some notes" } });
      expect(take.state().notes).toEqual("Some notes");
    });
  });

  describe("when clicking the remove button", () => {
    beforeEach(() => {
      take.find(".remove-btn").simulate("click");
    });

    it("calls the removeTake callback", () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  });
});