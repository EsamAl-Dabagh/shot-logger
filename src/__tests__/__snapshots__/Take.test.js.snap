// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Take
    confirmTake={[MockFunction]}
    id={1}
    removeTake={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          This is the TAKE component: 
          1
        </p>,
        <form>
          <TextField
            autoFocus={true}
            className="scene-input"
            label="Scene"
            onChange={[Function]}
            required={false}
            select={false}
            variant="outlined"
          />
          <TextField
            className="shot-input"
            label="Shot"
            onChange={[Function]}
            required={false}
            select={false}
            variant="outlined"
          />
          <TextField
            className="take-input"
            label="Take"
            onChange={[Function]}
            required={false}
            select={false}
            variant="outlined"
          />
          <TextField
            className="desc-input"
            label="Description"
            onChange={[Function]}
            required={false}
            select={false}
            variant="outlined"
          />
          <WithStyles(Checkbox)
            checked={false}
            className="goodTake-switch"
            color="primary"
            label="Good Take?"
            onChange={[Function]}
            value="goodTake"
          />
          <TextField
            className="notes-input"
            label="Notes"
            multiline={true}
            onChange={[Function]}
            required={false}
            select={false}
            variant="outlined"
          />
        </form>,
        <WithStyles(Button)
          className="remove-btn"
          color="secondary"
          onClick={[Function]}
          variant="contained"
        >
          <pure(DeleteIcon)
            className="delete-icon"
          />
        </WithStyles(Button)>,
        <WithStyles(Button)
          className="confirm-btn"
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          <pure(DoneIcon)
            className="confirm-icon"
          />
        </WithStyles(Button)>,
      ],
      "className": "take",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "This is the TAKE component: ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "This is the TAKE component: ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextField
              autoFocus={true}
              className="scene-input"
              label="Scene"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />,
            <TextField
              className="shot-input"
              label="Shot"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />,
            <TextField
              className="take-input"
              label="Take"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />,
            <TextField
              className="desc-input"
              label="Description"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />,
            <WithStyles(Checkbox)
              checked={false}
              className="goodTake-switch"
              color="primary"
              label="Good Take?"
              onChange={[Function]}
              value="goodTake"
            />,
            <TextField
              className="notes-input"
              label="Notes"
              multiline={true}
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "className": "scene-input",
              "label": "Scene",
              "onChange": [Function],
              "required": false,
              "select": false,
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "shot-input",
              "label": "Shot",
              "onChange": [Function],
              "required": false,
              "select": false,
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "take-input",
              "label": "Take",
              "onChange": [Function],
              "required": false,
              "select": false,
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "desc-input",
              "label": "Description",
              "onChange": [Function],
              "required": false,
              "select": false,
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "className": "goodTake-switch",
              "color": "primary",
              "label": "Good Take?",
              "onChange": [Function],
              "value": "goodTake",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "notes-input",
              "label": "Notes",
              "multiline": true,
              "onChange": [Function],
              "required": false,
              "select": false,
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <pure(DeleteIcon)
            className="delete-icon"
          />,
          "className": "remove-btn",
          "color": "secondary",
          "onClick": [Function],
          "variant": "contained",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "delete-icon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <pure(DoneIcon)
            className="confirm-icon"
          />,
          "className": "confirm-btn",
          "color": "primary",
          "onClick": [Function],
          "variant": "contained",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "confirm-icon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            This is the TAKE component: 
            1
          </p>,
          <form>
            <TextField
              autoFocus={true}
              className="scene-input"
              label="Scene"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />
            <TextField
              className="shot-input"
              label="Shot"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />
            <TextField
              className="take-input"
              label="Take"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />
            <TextField
              className="desc-input"
              label="Description"
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />
            <WithStyles(Checkbox)
              checked={false}
              className="goodTake-switch"
              color="primary"
              label="Good Take?"
              onChange={[Function]}
              value="goodTake"
            />
            <TextField
              className="notes-input"
              label="Notes"
              multiline={true}
              onChange={[Function]}
              required={false}
              select={false}
              variant="outlined"
            />
          </form>,
          <WithStyles(Button)
            className="remove-btn"
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            <pure(DeleteIcon)
              className="delete-icon"
            />
          </WithStyles(Button)>,
          <WithStyles(Button)
            className="confirm-btn"
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            <pure(DoneIcon)
              className="confirm-icon"
            />
          </WithStyles(Button)>,
        ],
        "className": "take",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "This is the TAKE component: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "This is the TAKE component: ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextField
                autoFocus={true}
                className="scene-input"
                label="Scene"
                onChange={[Function]}
                required={false}
                select={false}
                variant="outlined"
              />,
              <TextField
                className="shot-input"
                label="Shot"
                onChange={[Function]}
                required={false}
                select={false}
                variant="outlined"
              />,
              <TextField
                className="take-input"
                label="Take"
                onChange={[Function]}
                required={false}
                select={false}
                variant="outlined"
              />,
              <TextField
                className="desc-input"
                label="Description"
                onChange={[Function]}
                required={false}
                select={false}
                variant="outlined"
              />,
              <WithStyles(Checkbox)
                checked={false}
                className="goodTake-switch"
                color="primary"
                label="Good Take?"
                onChange={[Function]}
                value="goodTake"
              />,
              <TextField
                className="notes-input"
                label="Notes"
                multiline={true}
                onChange={[Function]}
                required={false}
                select={false}
                variant="outlined"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "className": "scene-input",
                "label": "Scene",
                "onChange": [Function],
                "required": false,
                "select": false,
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "shot-input",
                "label": "Shot",
                "onChange": [Function],
                "required": false,
                "select": false,
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "take-input",
                "label": "Take",
                "onChange": [Function],
                "required": false,
                "select": false,
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "desc-input",
                "label": "Description",
                "onChange": [Function],
                "required": false,
                "select": false,
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "className": "goodTake-switch",
                "color": "primary",
                "label": "Good Take?",
                "onChange": [Function],
                "value": "goodTake",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "notes-input",
                "label": "Notes",
                "multiline": true,
                "onChange": [Function],
                "required": false,
                "select": false,
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(DeleteIcon)
              className="delete-icon"
            />,
            "className": "remove-btn",
            "color": "secondary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "delete-icon",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(DoneIcon)
              className="confirm-icon"
            />,
            "className": "confirm-btn",
            "color": "primary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "confirm-icon",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
